### PR TITLE
Added notepad-app recipe

### PR DESCRIPTION
**Pull request by tahatrue**

### CODE REVIEW

**Comment by chbae** [comment]
Please use just one PR for notepad-app. https://github.com/GENIVI/meta-genivi-dev/pull/90 should be integrated in here.

I'll review it tonight. need to change a lot.

Thanks for new application contribution. :)

**Review by chbae — CHANGES_REQUESTED** [review]
At first, please remove unnecessary comment.

**Comment by gunnarx** [comment]
> At first, please remove unnecessary comment.

I agree.  Because we are short on time, whoever is merging this can do the fixup.

**Comment by DunderRoffe** [comment]
I have a cleaned up version on my fork:
https://github.com/dunderroffe/meta-genivi-dev/tree/patch-1

**Comment by zeenix** [comment]
@DunderRoffe cool, thanks. Can you create a PR?

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (chbae) [inline meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: Please add "SUMMARY" on top of this bb file.

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (chbae) [inline meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: You don't need this line because LICENSE is "CLOSED"

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (chbae) [inline meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: SRC_URI = "git://github.com/GENIVI/notepad-app.git;protocol=https"

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (chbae) [inline meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: Why this is RDEPENDS? I thought that it should be persistence-client-library if you are using this.

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (chbae) [inline meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: DEPENDS = "qtdeclarative"

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (chbae) [inline meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: /usr/share -> ${datadir}

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (chbae) [inline meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: /usr/share -> ${datadir}

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (chbae) [inline meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: /usr/share -> ${datadir}

**Review by chbae — CHANGES_REQUESTED** [review]
Thanks for integration but need to change somethig.

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (chbae) [reply meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: I checked notepad-app.pro file in your repository. Then you should move persistence-client-library to "DEPENDES"

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (chbae) [inline meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: # Copyright (C) 2017 GENIVI Alliance

**Comment by zeenix** [comment]
@DunderRoffe Can you please squash the fixes to recipe patch? Also a typo: Add recip**t**e summary

**Comment by chbae** [comment]
To install this application, please add this recipe into packagegroup or image.

**Comment by gunnarx** [comment]
I got kind of lost in the review feedback.  Anyway, it needed rebase so I now pushed a new version.

**Comment by gunnarx** [comment]
Build fails, not finding the .desktop file
09:51:43.066 | install: cannot stat '/var/lib/go-agent/pipelines/GDP-Yocto-QEMU_x86-64-meta-genivi-dev-GITHUB-PR/gdp/gdp-src-build/tmp/work/core2-64-poky-linux/notepad-app/1.0-r0/notepad-app.desktop': No such file or directory

**Comment by gunnarx** [comment]
Extra files to be installed, are they not normally under a files/ subdirectory?
@DunderRoffe what do you think?  Also, where would you prefer the .desktopfiles - as part of meta-genivi-dev or in the original application repos?

**Comment by DunderRoffe** [comment]
@gunnarx 
The .desktop files are preferably a part of meta-genivi-dev. They are specific for this platform which mean some apps wont have them in their upstream repo so then we need to put their .desktop files in meta-genivi-dev anyway. Hence, if all apps are handled this way we have a consistent way of doing it for all apps.

**Comment by DunderRoffe** [comment]
Updated it again. Was missing reference to where to find the .desktop file. Hopefully this works :)

**Comment by gunnarx** [comment]
Thanks, makes sense to me about the .desktop files.

New build error:
11:35:15.855 ERROR: notepad-app-1.0-r0 do_package_qa: QA Issue: notepad-app: The compile log indicates that host include and/or library paths were used.

Please check the log '/var/lib/go-agent/pipelines/GDP-Yocto-QEMU_x86-64-meta-genivi-dev-GITHUB-PR/gdp/gdp-src-build/tmp/work/core2-64-poky-linux/notepad-app/1.0-r0/temp/log.do_compile'

**Comment by gunnarx** [comment]
log.do_compile [attached](https://github.com/GENIVI/meta-genivi-dev/files/1004314/log.do_compile.txt)

**Comment by DunderRoffe** [comment]
While analyzing this log I found `cc1plus: warning: include location "/usr/include" is unsafe for cross-compilation [-Wpoison-system-directories]`. This warning is because notepad.pro file sets `INCLUDEPATH += /usr/include`. This is not supported while cross-compiling which makes the Quality Assurance in Yocto break the build while packaging .

**Comment by DunderRoffe** [comment]
While analyzing this log I found `cc1plus: warning: include location "/usr/include" is unsafe for cross-compilation [-Wpoison-system-directories]`. This warning is because notepad.pro file sets `INCLUDEPATH += /usr/include`. This is not supported while cross-compiling which makes the Quality Assurance in Yocto break the build while packaging .

I provided a patch for this locally but it should be upstreamed. However, since I do not have any setup that is able to compile this application outside Yocto I can not verify that this solution would work in the general case.

**Comment by DunderRoffe** [comment]
@chbae would re evaluate this PR? :)

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (chbae) [inline meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: why don't you contribute this patch to repository?

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (chbae) [inline meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: DEPENDS = "qtdeclarative persistence-client-library"

**Review by chbae — COMMENTED** [review]
need small change but LGTM.

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (DunderRoffe) [reply meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: Because I do not know if this patch would work for other solutions than the GDP Yocto build.

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (DunderRoffe) [reply meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: Isn't keeping a space a good practice since it makes it easier to do a "_append"?

Comment (chbae) [reply meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: @DunderRoffe 
Basically, all _append should be included space. So this is not good grammar. So I recommend that we can remove ".

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (chbae) [reply meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: Ok. I understand what you are thinking. This is workaround. I agree. Thanks

**Review by chbae — APPROVED** [review]

**Comment by chbae** [comment]
Tomorrow, I'll test this image and merge soon. If you test yourself, you can merge it. :)

**Comment by chbae** [comment]
I tested on my raspberrypi3 and it works fine but I can't see application icon in launcher.
@tahatrue Could you add icon?

**Review by chbae — CHANGES_REQUESTED** [review]
I tested on my raspberrypi3 and it works fine but I can't see application icon in launcher.
@tahatrue Could you add icon?

**Comment by DunderRoffe** [comment]
@chbae is persisting notes working for you on the RPi3? I tried it on the Minnowboard but did not get PCL to work.

**Comment by chbae** [comment]
@DunderRoffe
It works on my rpi3 but I can't show icon in launcher as I mentioned. I'll try it on minnowboard tomorrow. I'm sorry that what is PCL?

**Comment by DunderRoffe** [comment]
There is no icon so I made one will set up a PR to the notepad-app repo today.

**Comment by chbae** [comment]
@DunderRoffe Could you add icon and test again?
In this week, I'll join AGL. So I only can build and review.

**Comment by DunderRoffe** [comment]
Sorry @chbae never pushed the PR.
Here it is though: https://github.com/GENIVI/notepad-app/pull/2

**Comment by DunderRoffe** [comment]
The PR is now merged so I updated the recipe accordingly.

**Comment by gunnarx** [comment]
@DunderRoffe Good that we're running this update through PR checks because it seems to be failing. 

(By the way, my comment and signoff on this commit makes no sense.  In the contribution guidelines I later stated that it should not be done like that, so if you update and push again, please remove it before we merge to master.)  <-- I did this mistake, I'm just saying you can fix it if you're pushing a fix.

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (gunnarx) [inline meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: I think you might need a backslash here to prevent the line break from breaking the command.

**Review by gunnarx — CHANGES_REQUESTED** [review]

File: meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb
Comment (DunderRoffe) [reply meta-genivi-dev/recipes-dev-hmi/genivi-dev-platform-hmi/notepad-app.bb]: Ah yes you are right. I fix

**Comment by DunderRoffe** [comment]
Tested on Minnowboard and then saw that the icon did not appear although it was installed.
So I added small fix that properly locates the image. This should not effect the yocto builds

**Review by DunderRoffe — CHANGES_REQUESTED** [review]
Even though the application seems to be working fine with icon etc, something is wrong with the persistence system. Therefore this PR should not be merged until we have verified that the persisting is working as intended

**Comment by DunderRoffe** [comment]
In order to ease the process of merging meta-genivi-dev into genivi-dev-platform, I close and push this PR to a feature branch "notepadApp". It will be opened again when the reasons for not merging above are resolved.

**Pull request closed without merging**
